barcodes_search/tests/test_barcode_search.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017-Today GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class Tests(TransactionCase):
+    """Tests 'Barcodes Search'"""
+
+    def setUp(self):
+        super(Tests, self).setUp()
+        self.search_obj = self.env['barcode.search']
+        self.partner_with_barcode_id = self.ref(
+            'barcodes_search.partner_with_barcode')
+        self.rule_obj = self.env['barcode.rule']
+        # Add the key 'price' (and the according rule)
+        # because this key is added only if point of sale is installed
+        self.rule_obj._fields['type'].selection.append(('price', 'Price'))
+        self.price_rule = self.rule_obj.create({
+            'name': 'Demo Price Rule Barcode',
+            'barcode_nomenclature_id': self.ref(
+                'barcodes.default_barcode_nomenclature'),
+            'sequence': 1,
+            'type': 'price',
+            'encoding': 'ean13',
+            'pattern': '23.....{NNNDD}',
+        })
+
+    # Test Section
+    def test_01_search_single_result(self):
```

Comment:
Any reason you're numbering the tests? These are fine to be executed out of order IMO.

On a general note - if the tests are not fine to be executed out of order, it's worth a step back to see if you're properly setting the tests up. Typically unit tests should be focused on one goal per method and entirely atomic from all other methods. I haven't really found good use case for breaking those rules, but I'm sure there are some.